server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/container/ProjectAnalysisTaskContainerPopulator.java

```diff
@@ -136,7 +137,7 @@ public void populateContainer(TaskContainer container) {
    * List of all objects to be injected in the picocontainer dedicated to computation stack.
    * Does not contain the steps declared in {@link ReportComputationSteps#orderedStepClasses()}.
    */
-  private static List componentClasses() {
+  private static List<Object> componentClasses() {
```

Comment:
commit title "SONAR-8736 backdate issues that are caused by quality profile changes" is ambiguous.
"quality profile change" could both apply to swtiching from one QP to another to adding rule to a profile

could be "SONAR-8736 backdate issues caused by just enabled rules"